tests/next_tests/integration_tests/feature_tests/iterator_tests/test_horizontal_indirection.py

```diff
@@ -101,7 +101,7 @@ def test_direct_offset_for_indirection(program_processor):
 
     ref = np.zeros(shape)
     for i in range(shape[0]):
-        ref[i] = inp[i + cond[i]]
+        ref[i] = inp.ndarray[i + cond[i].item()]
```

Comment:
If getitem returns a field, it should be explicitly transformed first into a ndarray
```suggestion
        ref[i] = inp.ndarray[i + cond[i].ndarray.item()]
```

Reply:
```suggestion
        ref[i] = inp.ndarray[i + cond.ndarray[i]]
```